hedera-dedupe-bigquery/src/main/java/com/hedera/dedupe/DedupeRunner.java

```diff
@@ -0,0 +1,200 @@
+package com.hedera.dedupe;
+
+/*-
+ * ‌
+ * Hedera ETL
+ * ​
+ * Copyright (C) 2020 Hedera Hashgraph, LLC
+ * ​
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+import static com.hedera.dedupe.DedupeState.STATE_NAME_LAST_VALID_TIMESTAMP;
+import static com.hedera.dedupe.Utility.toBigQueryTimestamp;
+
+import java.time.Duration;
+import java.time.Instant;
+import lombok.extern.log4j.Log4j2;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+
+/**
+ * Dedupe runner periodically executes BigQuery job to deduplicate rows.
+ * State across runs is maintained using a separate BigQuery table.
+ * If a run exceeds the configured 'fixedRate', next run will be queued and started only after previous one finishes.
+ *
+ * Deduplication algorithm:
+ * (consensusTimestamp is unique for each transaction)
+ * 1. Get state: startTimestamp = lastValidTimestamp + 1 (from stateTable). 0 if not set.
+ * 2. Flag rows not in the streaming buffer: UPDATE table SET dedupe = 1 WHERE ...;
```

Comment:
Regarding this update: "UPDATE table SET dedupe = 1 WHERE ..."  - last time I tried it in BigQuery it would fail with an error something like "The rows affected by the update are in the streaming buffer, thus the query can't succeed". In other words, if the update is trying to change rows in a streaming buffer, BigQuery rejects such an update.  It's possible they changed the behaviour. Have you tried running it on a table with a streaming buffer.

Reply:
Holy shite. I haven't tested it with table with a streaming buffer. Would do early next week.
Right now working on automating deployment task. Might send parallel PRs.

Reply:
Tried a couple things today.
Created #12  to fix it as a followup (rather than part of 1000 lines PR).
Want to run the solution by you first before making code changes.